File: backend/components/metiers/administration_publique/tests/security/test_security.js
// test_security.js – Tests JS avancés de sécurité pour Threed
const { check_permission } = require('../utils/rbac');

describe('Sécurité Threed', () => {
  it('RBAC admin doit tout pouvoir', () => {
    expect(check_permission('admin', 'delete')).toBe(true);
  });
  it('RBAC guest ne peut pas delete', () => {
    expect(check_permission('guest', 'read')).toBe(true);
    expect(check_permission('guest', 'delete')).toBe(false);
  });
  it('doit protéger contre l’injection', () => {
    const user_input = "<script>alert('xss')</script>";
    const safe = user_input.replace(/</g, '').replace(/>/g, '');
    expect(safe.includes('<')).toBe(false);
    expect(safe.includes('>')).toBe(false);
  });
});

File: backend/components/metiers/administration_publique/tests/security/test_security.py
"""
Tests avancés de sécurité pour Threed (permissions, RBAC, audit, injection)
"""
from ..utils.rbac.rbac_utils import has_permission

def check_permission(role, permission):
    user = {'roles': [role]}
    return has_permission(user, permission)

def test_rbac_admin():
    assert check_permission('admin', 'delete')

def test_rbac_guest():
    assert check_permission('guest', 'read')
    assert not check_permission('guest', 'delete')

def test_injection_protection():
    # Simule une protection contre l'injection
    user_input = "<script>alert('xss')</script>"
    safe = user_input.replace('<', '').replace('>', '')
    assert '<' not in safe and '>' not in safe
